### DISCUSSION TITLE
Feature Request: .gdf() - to geopandas.GeoDataFrame

### DISCUSSION THREAD

**Discussion by raybellwaves** [duckdb/duckdb#14104] [Ideas]
Why do you want this feature?

I really like that duckdb has an export to pandas (https://duckdb.org/docs/guides/python/export_pandas.html).

I've started using duckdb for spatial analysis. I would really like a `.gdf()` which exports to a geopandas GeoDataFrame https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.html.

If there is interest i'd be happy to have a stab at putting in a PR for this. 

(similar discussions here: https://github.com/duckdb/duckdb/issues?q=is%3Aissue+geopandas+is%3Aclosed)

**Comment by CaptainInler** [comment]
`.gdf()` does not work (right now?) but there is an other easy way to transfer spatial data from DuckDB to geopandas.  
Here is an example:

```python
import duckdb
con = duckdb.connect()
con.install_extension("spatial")
con.load_extension("spatial")

# load some spatial data
con.install_extension("httpfs")
con.execute("LOAD httpfs")

con.execute("""
  CREATE OR REPLACE TABLE hoods AS
  SELECT *
  FROM st_read('https://duckdb-blobs.s3.us-east-1.amazonaws.com/nyc/hoods.shp');
""")
con.sql("DESCRIBE hoods").show()

# here the transfer to geopandas starts:
con.execute("CALL register_geoarrow_extensions()");

import geopandas as gpd
res = con.sql("select * from hoods");
gdf = gpd.GeoDataFrame.from_arrow(res)

gdf.describe()

```



Check out this presentation from the GeoPython 2025 Conference:
https://www.youtube.com/live/h7tIYkSIlYs?feature=shared&t=25054
The colab notebook used in that presentation:
https://colab.research.google.com/drive/1tPtzNhPbuP4gAE80wSvOEIpPnFr_jaJG#scrollTo=Qjg0mHxNhxms
